### PR TITLE
Bug 1147735 - Deprecate TelephonyCall.error r=drs

### PR DESCRIPTION
**Pull request by gabrielesvelto**

### CODE REVIEW

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=6b2d9c09a0d494cf900cd212026f3a1014e031b2

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=035fe6b4d26300c3424ccdf88e253f96382ddced

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=79b3792d0ef02743c47aaa15236428e736286094

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=b47721f861f1c8760ae9b480fe55b996494f47dd

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=87d5b9056ba2e4da0018ef0f6a8ac4961f06179b

File: apps/callscreen/js/calls_handler.js [hunk 1]
```diff
@@ -243,15 +243,15 @@ var CallsHandler = (function callsHandler() {
...
+   * automatically close the callscreen if no more calls are present.
    *
-   * @param evt {Object} The event delivered in the TelephonyCall.onerror
-   *        event-handler.
+   * @param evt {Object} The event delivered in the
+   *        TelephonyCall.ondisconnected event-handler.
    */
-  function handleBusyErrorAndPlayTone(evt) {
-    if (evt.call.error.name === 'BusyError') {
+  function playToneWhenBusy(evt) {
```
Comment (DouglasSherk) [inline apps/callscreen/js/calls_handler.js]: Why remove the ```handleBusyError``` part? That logic hasn't been removed.

File: apps/communications/dialer/js/telephony_helper.js [hunk 2]
```diff
@@ -182,19 +168,15 @@ var TelephonyHelper = (function() {
...
-        var messageType = emergencyOnly ? TelephonyMessages.NO_NETWORK :
-                                          TelephonyMessages.REGULAR_CALL;
-        TelephonyMessages.handleError(errorName, number, messageType);
+        TelephonyMessages.handleDisconnect(
+          evt.call.disconnectedReason, number, emergencyOnly
+        );
       });
-    };
+    });
+    call.addEventListener('disconnected', ondisconnected);
```
Comment (DouglasSherk) [inline apps/communications/dialer/js/telephony_helper.js]: It's awkward to add two event listeners like this. Combining them would be better.

File: apps/callscreen/js/calls_handler.js (hunk 1 above)
Comment (gabrielesvelto) [inline apps/callscreen/js/calls_handler.js]: I haven't removed it, just renamed to ``playToneWhenBusy`` to make the intent clearer.

File: apps/communications/dialer/js/telephony_helper.js (hunk 2 above)
Comment (gabrielesvelto) [inline apps/communications/dialer/js/telephony_helper.js]: Right, this is easily done.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=761d2bcc9a3c5936ea41c4d239699690b338665b

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9118c9eb8ff455f10e9df78b0069403a4cc4d057

File: apps/callscreen/js/calls_handler.js (hunk 1 above)
Comment (DouglasSherk) [inline apps/callscreen/js/calls_handler.js]: Yeah, my point was that I think the old name was more accurate. This isn't purely playing the busy tone -- it also has a side effect.

File: apps/emergency-call/test/unit/dialer_test.js [hunk 3]
```diff
@@ -96,10 +98,24 @@ suite('Emergency Dialer', function() {
...
       }).then(done, done);
     });
+
+    test('> displays an appropriate message on disconnected', function(done) {
+      this.sinon.spy(MockTelephonyMessages, 'handleDisconnect');
+      mockPromise.then(function() {
+        mockCall.ondisconnected({
+          call: { disconnectedReason: 'NormalCallClearing' }
+        });
+        sinon.assert.calledWith(
```
Comment (DouglasSherk) [inline apps/emergency-call/test/unit/dialer_test.js]: I think that we should get in the habit of asserting ```sinon.assert.calledOnce()``` as well.

File: apps/callscreen/js/calls_handler.js (hunk 1 above)
Comment (gabrielesvelto) [inline apps/callscreen/js/calls_handler.js]: Ah OK; I'll revert the change then.

**Comment by autolander** [comment]
Running tests
https://treeherder.mozilla.org/#/jobs?repo=gaia&revision=9d82e0887d98afa903c0c3861ff877696eb8fda1

**Pull request merged**